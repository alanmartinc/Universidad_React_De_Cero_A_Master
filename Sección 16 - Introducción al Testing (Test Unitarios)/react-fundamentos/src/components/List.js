import React from 'react'

const List = ({title, list}) => {
    return(
        <section>
            <h1 className='big'>{title}</h1>
            <ul>
                {list.map(item => (
                    <li key={item.id}>
                        {item.name}
                    </li>
                ))}
            </ul>
        </section>
    )
}

export default List