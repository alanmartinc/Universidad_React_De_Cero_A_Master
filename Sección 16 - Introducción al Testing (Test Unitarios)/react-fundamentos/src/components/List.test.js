import {configure, shallow, mount, render} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import List from './List'

configure({
    adapter: new Adapter()
})

describe('Probando componente <List/>', () => {
    test('Validar nodos', () => {
        const fruits = [
            {name: 'fresa', id: 1},
            {name: 'manzana', id: 2},
            {name: 'naranja', id: 3},
            {name: 'mango', id: 4}
        ]

        const wrapper = shallow(<List title='Frutas' list={fruits} />)

        // Validar si existe nodo
        expect(wrapper.find('h1').exists()).toBe(true)

        // Validar si tiene clase css
        expect(wrapper.find('h1').hasClass('big')).toBe(true)

        // Validar cantidad de elementos hijos
        expect(wrapper.find('ul').children().length).toBe(4)

        // Validar el contenido de texto
        console.log(wrapper.find('li').first().html())
        console.log(wrapper.find('li').last().html())

        expect(wrapper.find('ul').childAt(2).text()).toBe('naranja')

        // Validar tipo de nodo / elemento
        expect(wrapper.find('ul').childAt(2).type()).toBe('li')

        // Validar HTML
        expect(wrapper.find('ul').childAt(2).html()).toBe('<li>naranja</li>')
    })
})