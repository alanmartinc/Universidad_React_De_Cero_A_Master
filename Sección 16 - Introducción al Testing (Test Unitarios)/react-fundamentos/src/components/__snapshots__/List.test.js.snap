// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente <List/> Validar que coincida con Snapshot 1`] = `
<section>
  <h1
    className="big"
  >
    Frutas
  </h1>
  <ul>
    <li
      key="1"
    >
      fresa
    </li>
    <li
      key="2"
    >
      manzana
    </li>
    <li
      key="3"
    >
      naranja
    </li>
    <li
      key="4"
    >
      mango
    </li>
  </ul>
</section>
`;
